src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -76,12 +75,12 @@ public boolean hitExecutionNeeded(SearchContext context) {
     public void hitExecute(SearchContext context, HitContext hitContext) {
         Map<String, HighlightField> highlightFields = newHashMap();
         for (SearchContextHighlight.Field field : context.highlight().fields()) {
-            List<String> fieldNamesToHighlight;
+            Set<String> fieldNamesToHighlight;
```

Comment:
Why not add
```java
if (highlightFields.contains(fieldName)) { continue; }
```
around line 94?

That'll prevent two regexes that find the same field from highlighting it twice.